Test/QuickCheck/Modifiers.hs

```diff
@@ -343,4 +355,100 @@ instance (Arbitrary a, ShrinkState s a) => Arbitrary (Shrinking s a) where
 #endif /* NO_MULTI_PARAM_TYPE_CLASSES */
 
 --------------------------------------------------------------------------
+-- | @ASCII@.
+newtype ASCII = ASCII {getASCII :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+asciiGen :: Gen Char
+asciiGen = choose ('\0', '\127')
+
+instance Arbitrary ASCII where
+  arbitrary = ASCII `fmap` asciiGen
+
+--------------------------------------------------------------------------
+-- | @Latin1@: latin1 Char, same Arbitrary as Char itself.
+newtype Latin1 = Latin1 {getLatin1 :: Char}
```

Comment:
AllUnicode could be useful for code that handles surrogate pairs (and needs to detect mismatched pairs)...

`Text.pack`ing an invalid surrogate pair will just yield `U+FFFD`, so it's indeed not very useful for most common code. And I expect that code deailing with UTF-16 will most often deal directly with `Word`s, rather than `Char`s